nixos/modules/tasks/filesystems.nix

```diff
@@ -140,6 +140,15 @@ let
       (mkIf config.autoResize [ "x-systemd.growfs" ])
       (mkIf config.autoFormat [ "x-systemd.makefs" ])
       (mkIf (utils.fsNeededForBoot config) [ "x-initrd.mount" ])
+      (mkIf
+        # With scripted stage 1, depends is implemented by sorting 'config.system.build.fileSystems'
+        (config.depends != [ ] && (utils.fsNeededForBoot config -> _config.boot.initrd.systemd.enable))
```

Comment:
> and in every case except for _config.boot.initrd.systemd.enable, add these x-systemd.requires-mounts-for lines.

No, the only time we *don't* add the `x-systemd.requires-mounts-for=` option is when it's `fsNeededForBoot` and `boot.initrd.systemd.enable == false`, which is what your truth table shows. i.e. If the file system is mounted in scripted initrd, we let scripted initrd's semantics handle `depends` (which, as the comment indicates, is handled by the sorting of file systems).

> It's not clear immediately what the semantics of x-systemd.requires-mounts-for= (with nothing after it) is. That's the case when utils.fsNeededForBoot config is false

No it isn't. Re-parenthesizing, it's `"x-systemd.requires-mounts-for=${(optionalString (utils.fsNeededForBoot config) "/sysroot") + x}"`, i.e. the `x` string is unconditional and always included. There should never be a `x-systemd.requires-mounts-for=` with no value.

Reply:
> This is the first and only time that /sysroot is named in this file. Is there a place that could be linked to document or reference the structure of the initrd?

The only real documentation that talks about `/sysroot` is `man bootup`, but we reference `/sysroot` in several places for systemd initrd so I'm not sure why this particular instance needs a link to documentation when the others don't.

Reply:
> what your truth table shows

Yeah, I thought I had typed `== false` but obviously not. We're in agreement here about what's happening.

> Re-parenthesizing

That change makes this **significantly** clearer. I did not understand how this would render, as my commentary shows.